tests/integration.rs

```diff
@@ -6080,6 +6080,12 @@ mod common_parallel {
         let r = std::panic::catch_unwind(|| {
             // Resume the VM
             assert!(remote_command(&api_socket_restored, "resume", None));
+            // There is no way that we can ensure the 'write()' to the
+            // event file is completed when the 'resume' request is
+            // returned successfully, because the 'write()' was done
+            // asynchronously from a different thread of Cloud
+            // Hypervisor (e.g. the event-monitor thread).
+            thread::sleep(std::time::Duration::new(1, 0));
```

Comment:
How do we know that one second is enough?

Reply:
I have the same question :)
After looking at integration.rs, I found that there are some other similar sleep calls. Do we need to optimize that? @likebreath @rbradford

Reply:
It was my "educated" guess based on the nature of the racing. The racing here is essentially between the `http-api` thread (that is acknowledging the completion of the "resume" request) and the `event-monitor` thread (that is performing the `write()` to the event file). Unless we synchronize these two threads within Cloud Hypervisor, we have no way to avoid such racing from outside (e.g. from the integration test). Note that `read()` will be guaranteed to get the latest content only if it is invoked after "write()" competition.

Of course, I could be wrong and we can adjust accordingly based on our CI results in the future.

Reply:
> After looking at integration.rs, I found that there are some other similar sleep calls. Do we need to optimize that? 

You are right. There are (many) other hard-coded timeout for similar reasons and definitely good places for optimization to improve stability and efficiency. #5127 is one example for removing some of the timeouts. Ideas and contributions are welcome.

Reply:
The idea of using the events in the integration tests is to be able to wait for a specific event to be fired. Instead here this code doesn't wait for the events and instead checks they have fired. We should have a function call which we can call with : `wait_for_event(event_file, "resumed", EVENT_DEFAULT_TIMEOUT)`.

This is what #5127 is supposed to be about - adding and using a "wait until event" infrastructure.

Reply:
> The idea of using the events in the integration tests is to be able to wait for a specific event to be fired. 

This is true for #5217 but not for this particular integration test.

> Instead here this code doesn't wait for the events and instead checks they have fired.

Existing integration tests that call into `check_*_events_*` (including this one) are for "testing" the event monitor functionality, not for "using" the event monitor (say as a way to reduce hard-coded sleeps).

Adding a `wait_for_event` here is basically imposing a much longer timeout with busy waiting. I agree it could make this particular test more robust (because of the much longer waiting time), but at the expense of adding more CPU contention (which is the reason why this test is flaky now). 

I believe adding a short sleep would resolve this flaky issue and is a better choice for this particular case. Of course, I am open to different opinions.